### PR TITLE
Allow configuration of more OT options from Zephyr

### PR DESCRIPTION
**Pull request by markus-becker-tridonic-com**
Signed-off-by: Markus Becker <markus.becker@tridonic.com>

### CODE REVIEW

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -41,6 +41,16 @@ if(OT_BORDER_ROUTER)
     list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE=1")
 endif()
 
+option(OT_NCP "enable network co-processor NCP")
```
Comment (rlubos) [inline etc/cmake/options.cmake]: Did you prepare a PR towards the upstream OpenThread repository? Ideally, we should get this change from upstream, I'd rather avoid local changes to OT code base.

Please split this change into a separate commit, add `[ot fromlist]` prefix to the commit title and add a link to the upstream PR introducting this change in the commit message. We'll need to revert it before next upstream merge.

Comment (markus-becker-tridonic-com) [reply etc/cmake/options.cmake]: > Did you prepare a PR towards the upstream OpenThread repository? Ideally, we should get this change from upstream, I'd rather avoid local changes to OT code base.

No, I did not. Trying to get initial feedback on the complete package.

> Please split this change into a separate commit, add `[ot fromlist]` prefix to the commit title and add a link to the upstream PR introducting this change in the commit message. We'll need to revert it before next upstream merge.

Will do.

Thanks for immediate review. Please also have a look at https://github.com/zephyrproject-rtos/zephyr/pull/23249.

Comment (rlubos) [reply etc/cmake/options.cmake]: I will, I need to allocate more time for that though :)

Comment (markus-becker-tridonic-com) [reply etc/cmake/options.cmake]: Created https://github.com/openthread/openthread/pull/4633 in upstream OpenThread and rebased the branch.

**Comment by markus-becker-tridonic-com** [comment]
@rlubos upstream has merged in https://github.com/openthread/openthread/pull/4633 the equivalent of https://github.com/zephyrproject-rtos/openthread/pull/6/commits/e4927d973c22101a211f63633f3721965b15f515 . An update of OpenThread would be required to get the change in and drop e4927d973c22101a211f63633f3721965b15f515 . Are you doing that?

**Comment by rlubos** [comment]
@markus-becker-tridonic-com I was not planning to do an entire OT update, but on a second though it might be a good idea given the commit needed is already in. I'll give it a shot, in case there are no issues spotted with newer version I'll submit at PR with the update.

**Comment by rlubos** [comment]
Ok, so I've faced one issue with mbed TLS build, which needs a (trivial) fix upstream. I'll create a PR and after it's merged prepare an OpenThread update in Zephyr.

**Comment by rlubos** [comment]
@markus-becker-tridonic-com Your commit from OpenThread is now available on `zephyr` branch, please rebase.

**Comment by markus-becker-tridonic-com** [comment]
Done.

**Comment by rlubos** [comment]
CI on https://github.com/zephyrproject-rtos/zephyr/pull/23249 is green, merging.
